### PR TITLE
New l1 object

### PR DESCRIPTION
**Pull request by danielwinterbottom**
Switching to Riccardo's L1TObject class and producer

### CODE REVIEW

**Comment by adewit** [comment]
So before looking at this in detail, can you check the code compiles in all releases we intend to support? By only commenting 'please test' on the PR, a test build in all supported releases is triggered, when the test completes you can see whether they have passed, and the logs will be linked too. I can already see this PR will break compatibility with CMSSW 5_3_X, but I think @ajgilbert and @rcl11 are happy to drop support for this release. We should keep support for CMSSW_7_6 though, and this was actually already broken in your last PR, but unfortunately this was missed. If you run the test here you'll see where this comes from and you can fix it on this PR, rerunning the test until all checks come back ok.

**Comment by danielwinterbottom** [comment]
please test

**Comment by danielwinterbottom** [comment]
please test

**Comment by danielwinterbottom** [comment]
please test

File: plugins/ICL1TObjectProducer.hh [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+  edm::InputTag input_;
+  std::string branch_;
+  bool doBXloop_;
+  
+  std::vector<ic::L1TObject> *ic_l1t_object_;
+  edm::EDGetTokenT< BXVector<T> > m_EDToken_l1t_object;
+
+  // output file
+  edm::Service<TFileService> fs_;
```
Comment (adewit) [inline plugins/ICL1TObjectProducer.hh]: This isn't used, think we can remove this

File: plugins/ICL1TObjectProducer.hh [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+  
+  // tree
+  TTree * tree_;
```
Comment (adewit) [inline plugins/ICL1TObjectProducer.hh]: Same as the fileservice, can be removed I think as it isn't used, right?

File: Analysis/HiggsTauTau/filelists/Oct02_MC_80X_reHLT_VBFHToTauTau_M-125.dat [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+VBFHToTauTau_M125_13TeV_powheg_pythia8/crab_VBFHToTauTau/161002_110632/0000/EventTree_12.root
+VBFHToTauTau_M125_13TeV_powheg_pythia8/crab_VBFHToTauTau/161002_110632/0000/EventTree_40.root
+VBFHToTauTau_M125_13TeV_powheg_pythia8/crab_VBFHToTauTau/161002_110632/0000/EventTree_31.root
+VBFHToTauTau_M125_13TeV_powheg_pythia8/crab_VBFHToTauTau/161002_110632/0000/EventTree_34.root
+VBFHToTauTau_M125_13TeV_powheg_pythia8/crab_VBFHToTauTau/161002_110632/0000/EventTree_15.root
+VBFHToTauTau_M125_13TeV_powheg_pythia8/crab_VBFHToTauTau/161002_110632/0000/EventTree_11.root
+VBFHToTauTau_M125_13TeV_powheg_pythia8/crab_VBFHToTauTau/161002_110632/0000/EventTree_8.root
+VBFHToTauTau_M125_13TeV_powheg_pythia8/crab_VBFHToTauTau/161002_110632/0000/EventTree_30.root
+VBFHToTauTau_M125_13TeV_powheg_pythia8/crab_VBFHToTauTau/161002_110632/0000/EventTree_23.root
+VBFHToTauTau_M125_13TeV_powheg_pythia8/crab_VBFHToTauTau/161002_110632/0000/EventTree_4.root
```
Comment (adewit) [inline Analysis/HiggsTauTau/filelists/Oct02_MC_80X_reHLT_VBFHToTauTau_M-125.dat]: So just on these, I know there's only two relatively short filelists here so it's not really important, but could be worth putting them in a tarball anyway in the future. But up to you :-)

Comment (danielwinterbottom) [inline Analysis/HiggsTauTau/filelists/Oct02_MC_80X_reHLT_VBFHToTauTau_M-125.dat]: ok tarbaled these files and removed old lists

File: plugins/ICL1TObjectProducer.hh (hunk 2 above)
Comment (danielwinterbottom) [inline plugins/ICL1TObjectProducer.hh]: I've removed both these lines

**Pull request merged**